### PR TITLE
DX: check fixer's options for wording

### PR DESCRIPTION
**Pull request by kubawerlos**
Fixer `align_multiline_comment` currently has:

```
Configuration
-------------

``comment_type``
~~~~~~~~~~~~~~~~

Whether to fix PHPDoc comments only (``phpdocs_only``), any multi-line comment
whose lines all start with an asterisk (``phpdocs_like``) or any multi-line
comment (``all_multiline``).

Allowed values: ``'all_multiline'``, ``'phpdocs_like'`` and ``'phpdocs_only'``
```

Maybe better would be to update available option to `all`, `phpdoc_like` and `phpdoc_only`?

### CODE REVIEW

File: tests/Test/AbstractFixerTestCase.php [hunk 1]
```diff
@@ -494,7 +500,11 @@ private function getLinter(): LinterInterface
     private static function assertValidDescription(string $fixerName, string $descriptionType, string $description): void
     {
         self::assertMatchesRegularExpression('/^[A-Z`].+\.$/s', $description, sprintf('[%s] The %s must start with capital letter or a ` and end with dot.', $fixerName, $descriptionType));
-        self::assertStringNotContainsString('phpdocs', $description, sprintf('[%s] `PHPDoc` must not be in the plural in %s.', $fixerName, $descriptionType));
+        self::assertSame(
+            0,
+            substr_count($description, 'phpdocs') - substr_count($description, '`phpdocs_'), // allow the optio
```
Comment (keradus) [inline tests/Test/AbstractFixerTestCase.php:505]: optio

**Comment by keradus** [comment]
content of PR and text in PR opening are 2 separated things, do I understand right?

for changing options, why dropping plural?

**Comment by kubawerlos** [comment]
> why dropping plural?

It's not dropping, it still checks for plural, only with and exception fo having `phpdocs_*` in backticks, as `align_multiline_comment` has.

**Comment by keradus** [comment]
> > Allowed values: ``'all_multiline'``, ``'phpdocs_like'`` and ``'phpdocs_only'``
> 
> Maybe better would be to update available option to all, phpdoc_like and phpdoc_only?

I read it as you want to change the plurality

File: tests/Test/AbstractFixerTestCase.php [hunk 2]
```diff
@@ -494,7 +500,11 @@ private function getLinter(): LinterInterface
     private static function assertValidDescription(string $fixerName, string $descriptionType, string $description): void
     {
         self::assertMatchesRegularExpression('/^[A-Z`].+\.$/s', $description, sprintf('[%s] The %s must start with capital letter or a ` and end with dot.', $fixerName, $descriptionType));
-        self::assertStringNotContainsString('phpdocs', $description, sprintf('[%s] `PHPDoc` must not be in the plural in %s.', $fixerName, $descriptionType));
+        self::assertSame(
+            0,
+            substr_count($description, 'phpdocs') - substr_count($description, '`phpdocs_'), // allow the option to have "`phpdocs_*`"
```
Comment (keradus) [inline tests/Test/AbstractFixerTestCase.php:505]: I think right now we have _only_ `phpdocs_` (plural) as rule option, is it the case?

File: tests/Test/AbstractFixerTestCase.php [hunk 3]
```diff
@@ -494,7 +500,11 @@ private function getLinter(): LinterInterface
...
+            sprintf('[%s] `PHPDoc` must not be in the plural in %s.', $fixerName, $descriptionType),
+        );
```
Comment (keradus) [inline tests/Test/AbstractFixerTestCase.php:507]: I think right now we have _only_ `phpdocs_` (plural) as rule option, is it the case?

with that, maybe we should keep the opposite rule here, to have it plural for option?

or likely no - as we have only single rule having option like that, maybe we do not have enough options to decide if there should be a rule for it, sounds too early for me.


Original idea was to keep proper casing, which makes sense to have always the same casing for same, hm..., proper name. (polish "nazwa wlasna")

now , you suggest to unify using plural or singular, which really depends on the context and maybe should not be unified?

File: tests/Test/AbstractFixerTestCase.php [hunk 4]
```diff
@@ -464,6 +465,11 @@ protected function lintSource(string $source): ?string
     protected static function assertCorrectCasing(string $haystack, string $needle, string $fixerName, string $descriptionType): void
     {
         $exceptions = [
+            'PHPDoc' => [
+                'option:comment_type' => [
+                    'align_multiline_comment' => 2,
+                ],
+            ],
```
Comment (keradus) [inline tests/Test/AbstractFixerTestCase.php:472]: I dropped it on local machine to check how the error will look alike if someone would use `phpdocs_` option in other fixer.

Outcome is confusing:
```
1) PhpCsFixer\Tests\Fixer\Phpdoc\AlignMultilineCommentFixerTest::testXxxFixerConfigurationDefinitions
[align_multiline_comment] `PHPDoc` must be in correct casing in option:comment_type.
Failed asserting that 1 is identical to 3.
```

File: tests/Test/AbstractFixerTestCase.php (hunk 3 above)
Comment (kubawerlos) [reply tests/Test/AbstractFixerTestCase.php:507]: I'd say they should be unified, but let's start with not allowing more plural usages.

File: tests/Test/AbstractFixerTestCase.php (hunk 4 above)
Comment (kubawerlos) [reply tests/Test/AbstractFixerTestCase.php:472]: What is confusing in "`PHPDoc` must be in correct casing in option:comment_type."? It is quite clear to me what is wrong.

File: tests/Test/AbstractFixerTestCase.php (hunk 3 above)
Comment (keradus) [reply tests/Test/AbstractFixerTestCase.php]: only if whole option make sense when using singular form. not sure if that will be the case or not as we do not have many cases yet to validate that.

File: tests/Test/AbstractFixerTestCase.php (hunk 4 above)
Comment (keradus) [reply tests/Test/AbstractFixerTestCase.php]: it was confusing as test was complaining about plural/singular and test failure description was complaining about casing.

not the case anymore with newest state of this PR

**Review by keradus — APPROVED** [review]

**Pull request merged**